### ISSUE TITLE
Error when trying to use powershell 7.1 as default shell of ubuntu 20.04 in wsl

### ISSUE THREAD

**Issue by HCl-10** [PowerShell/PowerShell#14055]
## Steps to reproduce

Install powershell 7.1 in wsl ubuntu 20.04, then run the following command:

```sh
chsh -s /usr/bin/pwsh
```

Restart wsl, and an error will occur.

## Expected behavior

Start powershell in wsl ubuntu 20.04 normally.

## Actual behavior
An error occured:

```none
: 6: export: (x86)/Dev-Cpp/TDM-GCC-64/bin:/mnt/c/Program: bad variable name
```
(Maybe pwsh will load all Windows PATH before running wsl, `C:\Program Files (x86)\Dev-Cpp\TDM-GCC-64\bin` is the first in my PATH)

## Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.1.0
PSEdition                      Core
GitCommitId                    7.1.0
OS                             Linux 4.4.0-19041-Microsoft #488-Microsoft Mon Sep 01 13:43:00 PST 2020
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by fMichaleczek** [comment]
@HCl-10 I had problem with WSL and PATH which contain (Program Files) to compile OpenWrt, so I am sure it's a PowerShell Issue. see https://github.com/microsoft/WSL/issues/1640#issuecomment-713966706

**Comment by anmenaga** [comment]
@daxian-dbw may have more info on this.

**Comment by jonstelly** [comment]
Any ideas on this one?  I have the same issue.  I've got Ubuntu 20.04 WSL and the apt package install of powershell 7.1.X gives me the following error when configured as my shell: `: 6: export: (x86)/Microsoft: bad variable name`.

Oddly, if I install the dotnet global tool: `sudo dotnet tool update --tool-path /usr/local/bin powershell` and use that location as my shell I do not get this error.

I'd prefer to use the APT install, especially since the nuget tool package is a few days late with the 7.1.4 update and it would just feel more natural to use the APT install.

**Comment by TBBle** [comment]
As noted in #14640, the problem is probably that something is trying to
```
export PATH=...
```

and the `...` is not quoted correctly for the current shell, so the first thing that has a space before it, and doesn't look like a valid shell env-var name, i.e. `(x86)/Microsoft` or `(x86)/Dev-Cpp/TDM-GCC-64/bin:/mnt/c/Program`, causes this error.

I would _guess_ that PowerShell is being asked to evaluate a bash script, and differences in quoting behaviour (not a bug) are breaking this. So not really a PowerShell issue, unless reading the script is a built-in PowerShell behaviour.

I think it's much more likely it's a WSL issue, that (per https://github.com/microsoft/WSL/issues/1640#issuecomment-713966706) WSL's `appendWindowsPath` feature is assuming bash is the shell when it appends that path, and hence the quoting is incorrect when PowerShell is the shell.

**Comment by jonstelly** [comment]
Any suggestions of how I might debug this?  The error message isn’t super helpful.

Also, why the dotnet global tool install works but the apt package doesn’t?  I’m guessing some configuration that comes with the apt package but not with the global tool?

**Comment by TBBle** [comment]
_Assuming_ disabling `appendWindowsPath` avoids the issue (at the cost of not having Windows's `$PATH` in the WSL `$PATH`, then the problem is probably in the `appendWindowsPath` implementation, which might have magic to detect PowerShell, but it is failing for the APT-installed version.

It's also possible that the reason it works with the global tool version is that the global tool version has known issues in command-line parsing and quotes, e.g. #11747, and those issues coincide with the `appendWindowsPath` implementation to _work_ (or at least not-fail).

One other thing to check. Does `appendWindowsPath` actually _work_ with the gobal tool install? You've mentioned that it doesn't give the error, but I don't know if that means that `appendWindowsPath` is successful, or if it's just failing in a way that doesn't produce an error.

**Comment by jonstelly** [comment]
This seems to be caused by an issue in Ubuntu in `/etc/profile.d/apps-bin-path.sh`.  It sounds like apps-bin-path.sh has been fixed in 21.10 but Ubuntu on WSL is still 20.04.

Work-around: https://github.com/PowerShell/PowerShell/issues/9842#issuecomment-979325295
